sdk/src/types/block/payload/transaction/transaction_id.rs

```diff
@@ -1,11 +1,116 @@
 // Copyright 2020-2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-impl_id!(
-    pub TransactionId,
-    32,
-    "A transaction identifier, the BLAKE2b-256 hash of the transaction bytes. See <https://www.blake2.net/> for more information."
-);
+use crate::types::block::{output::OutputId, slot::SlotIndex, ConvertTo, Error};
 
+impl_id!(pub TransactionHash, 32, "The hash of a [`TransactionPayload`].");
+
+impl TransactionHash {
+    pub fn with_slot_index(self, slot_index: impl Into<SlotIndex>) -> TransactionId {
+        TransactionId {
+            hash: self,
+            slot_index: slot_index.into().to_le_bytes(),
+        }
+    }
+}
+
+/// A transaction identifier.
+#[derive(Clone, Copy, Eq, PartialEq, Hash, Ord, PartialOrd, Debug, packable::Packable)]
+#[packable(unpack_error = Error)]
+#[repr(C)]
+pub struct TransactionId {
+    pub(crate) hash: TransactionHash,
+    slot_index: [u8; core::mem::size_of::<SlotIndex>()],
+}
+
+impl TransactionId {
+    /// The length of a [`TransactionId`]
+    pub const LENGTH: usize = 40;
```

Comment:
This should be 36 since the SlotIndex is now u32 instead of u64

Reply:
now it is :)